### PR TITLE
Add the mono interpreter lane to CI

### PR DESCRIPTION
**Pull request by steveisok**
Provides the ability to set MONO_ENV_OPTIONS via msbuild prop (needed when running the interpreter).

### CODE REVIEW

File: eng/pipelines/libraries/run-test-job.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ parameters:
   isOfficialBuild: false
   liveRuntimeBuildConfig: ''
   runtimeFlavor: 'coreclr'
+  runtimeDisplayName: 'coreclr'
```
Comment (ViktorHofer) [inline eng/pipelines/libraries/run-test-job.yml]: why is this necessary? We already have runtimeFlavorName: https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/xplat-setup.yml#L53

File: src/libraries/sendtohelix.proj [hunk 2]
```diff
@@ -4,6 +4,9 @@
     <HelixBuild Condition="'$(HelixBuild)' == ''">$(BUILD_BUILDNUMBER)</HelixBuild>
     <HelixBuild Condition="'$(HelixBuild)' == ''">default</HelixBuild>
 
+    <!-- To enable the interpreter for mono, we need to pass an env switch -->
+    <MonoEnvOptions Condition="'$(MonoEnvOptions)' == '' and '$(MonoEnableInterpreter)' == 'true'">--interpreter</MonoEnvOptions>
```
Comment (ViktorHofer) [inline src/libraries/sendtohelix.proj]: nit: can you inline the variable. It doesn't seem worth putting that into a separate variable.

Comment (ViktorHofer) [reply src/libraries/sendtohelix.proj]: Ok ignore my comment, I didn't see that you are using that for different modes.

File: eng/pipelines/libraries/run-test-job.yml (hunk 1 above)
Comment (steveisok) [reply eng/pipelines/libraries/run-test-job.yml]: The way we're using runtimeFlavor seemed too broad.  We need a separate mono interpreter test lane, but 'interpreter' is not a runtime flavor.  If you add another and just keep it mono, then the jobs will fail b/c the names will be the same.

Comment (akoeplinger) [reply eng/pipelines/libraries/run-test-job.yml:10]: `runtimeDisplayName` feels like the wrong name to me, I'd rather call it something like `testConfig` which could be mono-interpreter, mono-fullaot, mono-...

**Comment by steveisok** [comment]
Vlad made a bunch of changes, so it's easier to open a new PR w/ just the CI parts.
